src/gui/properties/peerlistwidget.cpp

```diff
@@ -138,6 +136,67 @@ PeerListWidget::~PeerListWidget()
     delete m_copyHotkey;
 }
 
+void PeerListWidget::displayToggleColumnsMenu(const QPoint&)
+{
+    QMenu hideshowColumn(this);
+    hideshowColumn.setTitle(tr("Column visibility"));
+    QList<QAction*> actions;
+    for (int i = 0; i < PeerListDelegate::IP_HIDDEN; ++i) {
+        QAction *myAct = hideshowColumn.addAction(this->headerData(i, Qt::Horizontal, Qt::DisplayRole).toString());
+        myAct->setCheckable(true);
+        myAct->setChecked(!isColumnHidden(i));
+        actions.append(myAct);
+    }
+    int visibleCols = 0;
+    for (unsigned int i = 0; i < PeerListDelegate::COL_COUNT; i++) {
+        if (!isColumnHidden(i))
+            visibleCols++;
+
+        if (visibleCols > 1)
+            break;
+    }
+
+    // Call menu
+    QAction *act = hideshowColumn.exec(QCursor::pos());
+    if (act) {
+        int col = actions.indexOf(act);
+        Q_ASSERT(col >= 0);
+        Q_ASSERT(visibleCols > 0);
+        if (!isColumnHidden(col) && visibleCols == 1)
+            return;
+        qDebug("Toggling column %d visibility", col);
+        setColumnHidden(col, !isColumnHidden(col));
+        if (!isColumnHidden(col) && columnWidth(col) <= 5)
+            setColumnWidth(col, 100);
+        saveSettings();
+    }
+}
+
+QVariant PeerListWidget::headerData(int section, Qt::Orientation orientation, int role) const
+{
+    if (orientation == Qt::Horizontal) {
+        if (role == Qt::DisplayRole) {
+            switch(section) {
+            case PeerListDelegate::COUNTRY: return tr("Country");
+            case PeerListDelegate::IP: return tr("IP");
+            case PeerListDelegate::PORT: return tr("Port");
+            case PeerListDelegate::CONNECTION: return tr("Connection");
+            case PeerListDelegate::FLAGS: return tr("Flags");
+            case PeerListDelegate::CLIENT: return tr("Client", "i.e.: Client application");
+            case PeerListDelegate::PROGRESS: return tr("Progress", "i.e: % downloaded");
+            case PeerListDelegate::DOWN_SPEED: return tr("Down Speed", "i.e: Download speed");
+            case PeerListDelegate::UP_SPEED: return tr("Up Speed", "i.e: Upload speed");
+            case PeerListDelegate::TOT_DOWN: return tr("Downloaded", "i.e: total data downloaded");
+            case PeerListDelegate::TOT_UP: return tr("Uploaded", "i.e: total data uploaded");
+            case PeerListDelegate::RELEVANCE: return tr("Relevance", "i.e: How relevant this peer is to us. How many pieces it has that we don't.");
+            default:
+                return QVariant();
```

Comment:
Although we assume to have `return` statement in the same line in conditional expressions. I think the same rule can be applied in such cases. @sledgehammer999?